src/OpenSage.Game/Logic/Object/Update/AIUpdate/SupplyAIUpdate.cs

```diff
@@ -0,0 +1,222 @@
+﻿using System;
+using System.Numerics;
+using OpenSage.Content;
+using OpenSage.Data.Ini;
+using OpenSage.Mathematics;
+
+namespace OpenSage.Logic.Object
+{
+    public abstract class SupplyAIUpdate : AIUpdate
+    {
+        public GameObject CurrentSupplyTarget;
+        private SupplyCenterDockUpdate _currentTargetDockUpdate;
+        public SupplyGatherStates SupplyGatherState;
+
+        public enum SupplyGatherStates
+        {
+            DEFAULT,
```

Comment:
Why the shouty-caps here? 😄 We usually use PascalCase for enum values.

Reply:
Sorry, I am used to it from work :D -> done